prime-router/src/main/kotlin/azure/BatchFunction.kt

```diff
@@ -103,7 +103,9 @@ class BatchFunction(
                     workflowEngine.recordAction(actionHistory)
                 }
             } else {
-                workflowEngine.handleBatchEvent(event, maxBatchSize, backstopTime) { headers, txn ->
+                workflowEngine.handleBatchEvent(event, maxBatchSize, backstopTime, !receiver.topic.isUniversalPipeline)
```

Comment:
I'm not too sure about this solution, it seems kind of unmaintainable to me, given the possibility for having to support other pipelines in the future. Generally, we should stay away sprinkling pipeline-based conditionals throughout the code. We have the` !receiver.topic.isUniversalPipeline` check here and then in the trailing lambda we do:
```
if (receiver.topic.isUniversalPipeline) {
    batchUniversalData(validHeaders, actionHistory, receiver, txn)
}
```

Why don't we just update `batchUniversalData` to NOT download the files again? Furthermore, why does handleBatchEvent only need to run for universal pipeline? There seem to be deeper issues here that we should look into/resolve.